### ISSUE TITLE
update TS Colin to use current version of geoknife

### ISSUE THREAD

**Issue by jread-usgs** [USGS-OWI/owi-blog#191]
https://github.com/USGS-OWI/owi-blog/blob/master/content/tsColinPrecip.Rmd

See discussion captured here and solution mentioned by Alison for how this can be done w/ the current version. https://github.com/USGS-R/geoknife/issues/278

**Comment by lindsaycarr** [comment]
thanks for making an issue for this :) I just had it on my task list.

**Comment by jread-usgs** [comment]
Tropical Storm Colin (TS Colin) made landfall on June 6 in western Florida. The storm moved up the east coast, hitting Georgia, South Carolina, and North Carolina. We can explore the impacts of TS Colin using open data and R. Using the USGS-R `geoknife` package, we can pull precipitation data by county. 

### First, we created two functions. One to fetch data and one to map data.

Function to retrieve precip data using [`geoknife`](https://github.com/USGS-R/geoknife):

```r
getPrecip <- function(states, startDate, endDate){
  
  # fips webgeom
  wg_f <- webgeom(geom = 'derivative:US_Counties', attribute = 'FIPS')
  
  county.fips <- maps::county.fips %>% 
    tidyr::separate(polyname, c('statename', 'county'), ',') %>%
    filter(statename %in% states)
  
  stencil <- webgeom(geom = 'derivative:US_Counties',
                     attribute = 'FIPS',
                     values = counties_fips$fips)
  
  fabric <- webdata(url = 'http://cida.usgs.gov/thredds/dodsC/stageiv_combined', 
                    variables = "Total_precipitation_surface_1_Hour_Accumulation", 
                    times = c(as.POSIXct(startDate), 
                              as.POSIXct(endDate)))
  
  job <- geoknife(stencil, fabric, wait = TRUE, REQUIRE_FULL_COVERAGE=FALSE)
  check(job)
  return( result(job, with.units=TRUE))
  
}
```

Function to map cumulative precipitation data using R package `maps`:

```r
precipMap <- function(precipData, startDate, endDate){
 ...
```

### Now, we can use those functions to fetch data for specific counties and time periods.

TS Colin made landfall on June 6th and moved into open ocean on June 7th. Use these dates as the start and end times in our function (need to account for timezone, +5 UTC). We can visualize the path of the storm by mapping cumulative precipitation for each county. 

```r
library(dplyr)
library(tidyr)
library(geoknife) #order matters because 'query' is masked by a function in dplyr
library(RColorBrewer)
library(maps)

statesTSColin <- c('florida', 'alabama', 'georgia', 'south carolina', 'north carolina')
startTSColin <- "2016-06-06 05:00:00"
endTSColin <- "2016-06-08 05:00:00"

precipData <- getPrecip(states = statesTSColin, 
                        startDate = startTSColin, 
                        endDate = endTSColin)
precipMap(precipData) # can't it get start and end date from the data?
```

# Questions

Please direct any questions or comments on `geoknife` to:
[https://github.com/USGS-R/geoknife/issues](https://github.com/USGS-R/geoknife/issues)

*Edited on 11/22. Works with geoknife v1.4.0 on CRAN. Changes to fips retrieval to use maps package [this geoknife issue](https://github.com/USGS-R/geoknife/issues/278).*

**Comment by jread-usgs** [comment]
@lindsaycarr I made some suggestions. Hard to do in your PR because you weren't making changes to this section

**Comment by jread-usgs** [comment]
oh, and I think you need to zero pad the fips in the change I made☝️ ?

**Comment by lindsaycarr** [comment]
So I implemented those changes yesterday, but I have to change the plotting code because it needs the state and county names to make the maps. When I ran this new stuff, it did not give me the plot I would expect. Committing soon

**Comment by lindsaycarr** [comment]
Closed via #204
